python/samples/demos/graphrag/graphrag_service.py

```diff
@@ -0,0 +1,230 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from collections.abc import AsyncGenerator
+from typing import Any, Literal
+
+import pandas as pd
+import yaml
+
+import graphrag.api as api
+from graphrag.config.create_graphrag_config import GraphRagConfig, create_graphrag_config
+from graphrag.index.typing import PipelineRunResult
+from semantic_kernel.connectors.ai import PromptExecutionSettings
+from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
+from semantic_kernel.contents import AuthorRole, ChatHistory, ChatMessageContent, StreamingChatMessageContent
+
+logger = logging.getLogger(__name__)
+
+
+class GraphRagPromptExecutionSettings(PromptExecutionSettings):
+    response_type: str = "Multiple Paragraphs"
+    search_type: Literal["local", "global", "drift"] = "global"
+
+
+class GraphRagChatCompletion(ChatCompletionClientBase):
```

Comment:
Given that this is a demo/sample, I would suggest, as always, to (over)comment this file. You can simply ask the model to do that for you to save some time...